src/main/java/com/bringframework/configurator/BoboValueAnnotationConfiguration.java

```
@@ -0,0 +1,73 @@
+package com.bringframework.configurator;
+
+import com.bringframework.BoboFactory;
+import com.bringframework.annotation.BoboValue;
+import com.bringframework.exception.BoboException;
+import com.bringframework.util.TypeResolverUtil;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.net.URL;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+public class BoboValueAnnotationConfiguration implements BoboConfigurator {
+
+    private final Map<String, String> propertiesMap;
+
+    private final String KEY_VALUE_DELIMITER = "=";
+
+    private final String DEFAULT_PROPERTIES_FILE_NAME = "application.properties";
+
+    public BoboValueAnnotationConfiguration() {
+        propertiesMap = findAvailableProperties()
+                .stream()
+                .map(line -> line.split(KEY_VALUE_DELIMITER))
+                .collect(toMap(key -> key[0].trim(), value -> value[1].trim()));
+    }
+
+    @Override
+    public void configure(Object bobo, BoboFactory registry) {
+        for (Field field : bobo.getClass().getDeclaredFields()) {
+            if (field.isAnnotationPresent(BoboValue.class)) {
```

How about checking whether this annotated field has a number or string type? And if the field is not a number or string then we can throw an exception. 
Can you check similar logic in Spring?

---

Already covered